src/c/metadata.c

```diff
@@ -43,8 +44,14 @@ edgex_deviceprofile *edgex_metadata_client_get_deviceprofile
     ename
   );
 
+  iot_data_t *jwt_data = edgex_secrets_request_jwt (secretprovider);  
```

Comment:
Are we requiring that JWT authentication always be used, going forward? 
Or does edgex_secrets_request_jwt() return a harmless result if not using JWT authentication?

Reply:
For outgoing HTTP requests (to metadata):  Query secret store for a JWT, include JWT in outgoing requests if present.  (Existing HTTP request generator already sends JWT on outgoing requests, but metadata service stub doesn't provide one prior to this implementation).  For security-disabled, no JWT will be returned and then not sent.  For security-enabled will get it from vault and existing implementation will send it.

For incoming HTTP requests: Authorize all inbound requests except for PING.  When security is disabled, the JWT verification function always succeeds.  When security is enabled, forwards the request to vault.

Current draft PR supply dummy token instead of no token (will fix in next update).  Also current implementation does not renew the vault token (tech debt that needs to be fixed to enable the feature).